app/_hub/kong-inc/request-callout/overview/_index.md

```diff
@@ -0,0 +1,298 @@
+headers to be forwarded as callout request headers.
+
+#### A callout that retries a failed request
+
+```
+{
+	"name": "request-callout",
+	"config": {
+		"callouts": [
+			{
+				"name": "auth1",
+				"request": {
+					"url": "https://run.mocky.io/v3/3d380788-e25f-4101-970b-70e95eacf892",
+					"method": "GET",
+					"query": {
+						"forward": true
+					},
+					"error": {
+						"http_statuses": [500, 502, 503],
+						"retries": 3
+					}
+				},
+				"response": {
+					"body": {
+						"store": true,
+						"decode": true
+					}
+				}
+			}
+		],
+		"upstream": {
+			"headers": {
+				"custom": {
+					"Authorization": "Bearer $(callouts.auth1.response.body.token)"
+				}
+			}
+		}
+	}
+}
+```
+
+Note the presence of an `error` field, specifying the status codes considered 
+errors and the number of retries to make. The schema reference contains the full 
+list of supported configurations.
+
+#### A callout that caches the result
+
+```
+{
+	"name": "request-callout",
+	"config": {
+		"callouts": [
+			{
+				"name": "auth1",
+				"request": {
+					"url": "https://run.mocky.io/v3/3d380788-e25f-4101-970b-70e95eacf892",
+					"method": "GET",
+					"query": {
+						"forward": true
+					},
+					"error": {
+						"http_statuses": [500, 502, 503],
+						"retries": 3
+					}
+				},
+				"response": {
+					"body": {
+						"store": true,
+						"decode": true
+					}
+				}
+			}
+		],
+		"upstream": {
+			"headers": {
+				"custom": {
+					"Authorization": "Bearer $(callouts.auth1.response.body.token)"
+				}
+			}
+		},
+		"cache": {
+			"strategy": "memory",
+			"cache_ttl": 360
+		}
+	}
+}
+```
+
+This will result in the callout response being cached for 360 seconds. Each 
+callout object may also contain a `cache` component. The schema reference
+contains the full list of supported configurations.
+
+#### A "Request Collapsing" callout
+
+A callout can be built to issue requests to multiple APIs and build a response 
+from Kong.
+
+```
+{
+	"name": "request-callout",
+	"config": {
+		"callouts": [
+			{
+				"name": "c1",
+				"request": {
+					"url": "http://httpbin.org/uuid",
+					"method": "GET"
+				},
+				"response": {
+					"body": {
+						"decode": true
+					}
+				}
+			},
+			{
+				"name": "c2",
+				"request": {
+					"url": "http://httpbin.org/anything",
+					"method": "GET"
+				},
+				"response": {
+					"body": {
+						"decode": true
+					}
+				}
+			}
+		],
+		"upstream": {
+			"by_lua": "kong.response.exit(200, { uuid = kong.ctx.shared.callouts.c1.response.body.uuid, origin = kong.ctx.shared.callouts.c2.response.body.url})"
+		}
+	}
+}
+```
+
+In this toy example, we are taking one field of the first callout, another from 
+the second and building a full response.
+
+## The callout context
+
+Callout request and response context is stored in 
+`kong.ctx.shared.callouts.<name>`. The request context contains:
+- `.<name>.params`: the full config for the callout request, including `url`, 
```

Comment:
I find `.<name>.params` misleading because it is accessible via `.<name>.request.params` but user might think it is `.<name>.params`. Same below.